### DISCUSSION TITLE
Vertically and horizontally centered content in slides (revealjs)

### DISCUSSION THREAD

**Discussion by awav** [quarto-dev/quarto-cli#2951] [Q&A] [answered]
Hello everyone,

Can anyone advise how to place content on all slides (or one slide) such that text and images would be in the center? I've tried suggestions from here https://github.com/quarto-dev/quarto-cli/discussions/1386, and they don't work at all. I've mananged to center an image and the text only by using `::: {style="text-align: center;}`.

Thanks!

**Comment by mcanouil** [comment, marked as answer]
#1386 is for centering content of a column, not the content of a slide.

Inspired from @pat-s post on Twitter[^1]:

````
---
format:
  revealjs:
    include-in-header: 
      text: |
        <style>
        .center-xy {
          margin: 0;
          position: absolute;
          top: 50%;
          left: 50%;
          -ms-transform: translateY(-50%), translateX(-50%);
          transform: translateY(-50%), translateX(-50%);
        }
        </style>
---

## Slide 1

::: {.center-xy}

Text

:::

````

[^1]: <https://twitter.com/patscli/status/1582832917302358016>

**Reply by RoyiAvital** [reply]
@mcanouil , Should it work on images as well?

**Reply by mcanouil** [reply]
@RoyiAvital Simply try it 😉

On a side note: please avoid commenting and asking the same thing on multiple threads to several persons without trying the solution you found first. This is just causing extra work for us in addition to diluting the thread which has a working answer.

**Reply by RoyiAvital** [reply]
@mcanouil , It was a polite way to say I tried and it didn't work.
So I asked if it should work.

I am struggling with having an image take 90% of the width of the slide and being vertically aligned.

**Reply by mcanouil** [reply]
>  It was a polite way to say I tried and it didn't work.
So I asked if it should work.

Then, you need to state what you tried (by providing a complete and small Quarto document) and what did not work directly to avoid this kind of back and forth.
(It works, although, at least with the latest version, the figure top-left corner is at the center)

Also in the reply there is a link to another issue with variants to achieve vertical centering.

**Reply by mcanouil** [reply]
Directly derived from <https://github.com/quarto-dev/quarto-cli/discussions/1386>.
Unfortunately, this is HTML/CSS and not markdown, so at some point it will be useful to you to learn the basics of HTML/CSS (see <https://www.w3schools.com/Css/> for example).

<table>
<tr><th>Quarto document</th><th>HTML</th></tr>
<tr>
<td>

````qmd
---
format:
  revealjs:
    include-in-header: 
      text: |
        <style>
        .v-center-container {
          display: flex;
          justify-content: center;
          align-items: center;
          height: 90%;
        }
        </style>
---

## Slide 1

::: {.v-center-container}

![](https://quarto.org/quarto.png)

:::

````

</td>
<td>
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/0be7ed5f-9e65-480a-b47e-a39a241e7fe2">
</td>
</tr>
</table>

**Reply by RoyiAvital** [reply]
@mcanouil , This actually worked very well!

The only issue is the width of the image can not be controlled.
I have `SVG` file, I'd like it to have a width of `90%`. 

I tried:

```qmd
## Test SVG

::: {style="display: flex; justify-content: center; align-items: center; height: 90%;"}

![](MemoryLayout.svg){width="90%"}

:::
```

It didn't seem to work though.
Is it the `flex` thing?

**Reply by mcanouil** [reply]
Again, you need to provide a complete and fully reproducible example.
Look at what I shared, you can copy/paste it and see the results.
We can't with what you shared without having to modify it hopefully the same way as your document that did not work.

PS: no need to ping/tag people within a thread and at every answer. We are watching basically everything, so you simply make your notifications more prominent for no clear reasons.
It's a bit like yelling someone's name in the middle of the conversation with that same person, in some aspects.

**Reply by RoyiAvital** [reply]
No need for new code, on your code, can you set the image width?

What I tried, and didn't work, is:

```qmd
---
format:
  revealjs:
    include-in-header: 
      text: |
        <style>
        .v-center-container {
          display: flex;
          justify-content: center;
          align-items: center;
          height: 90%;
        }
        </style>
---

## Slide 1

::: {.v-center-container}

![](https://quarto.org/quarto.png){width="90%"}

:::
```

**Reply by mcanouil** [reply]
It's because of the flex component, you can't use relative width, because it is no longer relative to the whole slide, but to the `.v-center-container` div, so you have to use absolute dimensions.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/205a7751-6f79-45ec-8834-eeec1e82def0">

**Reply by RoyiAvital** [reply]
I [saw](https://stackoverflow.com/questions/72941210) a nice trick:

```qmd
## Vertically Aligned 

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam luctus ultrices
leo sit amet accumsan.

::: {layout="[[-1], [1], [-1]]"}

![](https://placeholder.pics/svg/200){fig-align="center"}

:::
```

Yet it doesn't seem to work on the latest version.
The image is still not vertically centered.

**Reply by RoyiAvital** [reply]
I think I found something:

```qmd
![](https://placeholder.pics/svg/200){.absolute left=0 right=0 bottom=0 top=0 height="90%" style="margin: auto auto;"}
```
For the above, one can set the height factor while the image will be horizontally and vertically centered relatively to the slide.

Based on [How to Center an Absolute Positioned Element Vertically and Horizontally with CSS](https://www.freecodecamp.org/news/how-to-center-an-absolute-positioned-element).

**Comment by JonSilas** [comment]
I think this is out of date now, I just use the following. 

```
---
format:
  revealjs: 
    center: true
---

## Slide 1

Text
```

**Reply by mcanouil** [reply]
No, it isn't.
This option is centering everything.

| https://github.com/quarto-dev/quarto-cli/discussions/2951#discussioncomment-3927297 | https://github.com/quarto-dev/quarto-cli/discussions/2951#discussioncomment-6441261 |
|--------|--------|
| <img width="729" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/b534e33b-3bec-4af2-ad70-3851a750e90d"> | <img width="729" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/4a5127b3-9aa4-44d4-b1cf-bd77b1a629e8"> |

**Reply by JonSilas** [reply]
Quite right! My mistake.

**Comment by chuxinyuan** [comment]
> #1386 is for centering content of a column, not the content of a slide.
> 
> Inspired from @pat-s post on Twitter[1](#user-content-fn-1-47fea18becbc2ec2d7ef6b1de56dc0af):
> 
> ```
> ---
> format:
>   revealjs:
>     include-in-header: 
>       text: |
>         <style>
>         .center-xy {
>           margin: 0;
>           position: absolute;
>           top: 50%;
>           left: 50%;
>           -ms-transform: translateY(-50%), translateX(-50%);
>           transform: translateY(-50%), translateX(-50%);
>         }
>         </style>
> ---
> 
> ## Slide 1
> 
> ::: {.center-xy}
> 
> Text
> 
> :::
> ```
> 
> ## Footnotes
> 
>     1. https://twitter.com/patscli/status/1582832917302358016 [↩](#user-content-fnref-1-47fea18becbc2ec2d7ef6b1de56dc0af)

1. If `number-sections: true`, how can I keep this slide unaffected?
2. This has no effect on font size，why？

```
    include-in-header: 
      text: |
        <style>
        .font {
          font-size: 150px;
        }
        </style>
```

**Reply by mcanouil** [reply]
@chuxinyuan could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by chuxinyuan** [reply]
````qmd
---
title: "Quarto slides"
lang: zh
format:
  revealjs: 
    number-sections: true
    number-depth: 3
    include-in-header: 
      text: |
        <style>
        .font {
          font-size: 200px;
        }
        .center-xy {
          margin: 0;
          position: absolute;
          top: 50%;
          left: 50%;
          -ms-transform: translateY(-50%), translateX(-50%);
          transform: translateY(-50%), translateX(-50%);
        }
        </style>
---


## 致谢 {number-sections="false"}

::: {.center-xy}
谢谢
:::
````

1. `font-size: 200px` Have no effect to font.
2.  I want to remove "1 致谢".

**Reply by mcanouil** [reply]
Could you format your example exactly as I showed?
